### PR TITLE
Feature/misc improvements

### PR DESCRIPTION
**Pull request by aheritier**
Various bug fixes and improvements

screenshot 2015-09-24 22 54 43
screenshot 2015-09-24 22 54 31

Live demo: https://aheritier.ci.cloudbees.com/disk-usage-simple/

### CODE REVIEW

**Comment by aheritier** [comment]
Please @reviewbybees (cc @recena @ndeloof) review the result (5 classes/6 jelly files)

**Comment by recena** [comment]
@aheritier I'm sorry, I didn't have chance to review it, was merged.

**Comment by aheritier** [comment]
I think it doesn't avoid to do a review. You can comment code and and commits directly. I'll review all feedbacks

File: src/main/java/com/cloudbees/simplediskusage/JobDiskItem.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import hudson.model.Job;
+
+/**
+ * A job directory path on the disk with its usage information
+ */
+public class JobDiskItem extends DiskItem {
+    final String fullName;
+    final String url;
+
+    public JobDiskItem(Job job, Long size) {
```
Comment (recena) [inline src/main/java/com/cloudbees/simplediskusage/JobDiskItem.java]: `Job<?, ?> job`

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+public class QuickDiskUsageInitializer {
+    /**
+     * Let's update data after Jenkins is started and all jobs are loaded
+     */
+    @Initializer(after = InitMilestone.JOB_LOADED)
+    public static void initialize() {
+        // TODO switch to Jenkins.getActiveInstance() once 1.590+ is the baseline
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins == null) {
```
Comment (recena) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java]: `Jenkins.getActiveInstance()` and remove this `if`

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            throw new IllegalStateException("Jenkins has not been started, or was already shut down");
+        }
+        QuickDiskUsagePlugin plugin = jenkins.getPlugin(QuickDiskUsagePlugin.class);
+        if (plugin == null) return;
+        plugin.refreshData();
```
Comment (recena) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java]:
```
if (plugin == null) {
    return;
}
```

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageManagementLink.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    /**
+     * Only sysadmin can access this page.
+     */
+    @Override
+    public Object getTarget() {
+        // TODO switch to Jenkins.getActiveInstance() once 1.590+ is the baseline
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins == null) {
+            throw new IllegalStateException("Jenkins has not been started, or was already shut down");
```
Comment (recena) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageManagementLink.java]: Ditto

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+
+@Extension
+@Singleton
+public class QuickDiskUsagePlugin extends Plugin {
+
+    public static final String DISK_USAGE =
+            System.getProperty("os.name").toLowerCase().contains("mac")
+                    ? "du -ks" // OSX doesn't have ionice, this is only used during dev on my laptop
+                    : "ionice -c 3 du -ks";
+
```
Comment (recena) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java]: We should specify on the [wiki page](https://wiki.jenkins-ci.org/display/JENKINS/CloudBees+Simple+Disk+Usage+Plugin) its platform compatibility.

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+    private long lastRunEnd = 0;
+
+    @Override
+    public void start() throws Exception {
+        try {
+            load();
+        } catch (IOException e) {
+            logger.log(Level.WARNING, "Failed to load "+getConfigXml(),e);
+        }
+        if(isRunning()){
```
Comment (recena) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java]: `if (isRunning()) {`

spaces according the rest of the files.

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+                logger.log(Level.INFO, "Unable to run disk usage check", e);
+                lastRunEnd = lastRunStart;
+            } finally {
+                SecurityContextHolder.setContext(impersonate);
+            }
+            try{
+                // Save data
+                save();
+            } catch (IOException e) {
+                logger.log(Level.WARNING, "Failed to save "+getConfigXml(),e);
```
Comment (recena) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java]: `logger.log(Level.WARNING, "Failed to save " + getConfigXml(), e);`

spaces

File: src/main/resources/com/cloudbees/simplediskusage/QuickDiskUsagePlugin/index.jelly [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+                                ${e.usage}
+                            </td>
+                            <td class="pane" style="text-align: right">
+                                <s:link href="clean/?job=${e.fullName}" post="true">run cleanup</s:link>
+                            </td>
+                        </tr>
+                    </j:forEach>
+                </tbody>
+            </table>
+            <p></p>
```
Comment (recena) [inline src/main/resources/com/cloudbees/simplediskusage/QuickDiskUsagePlugin/index.jelly]: Why this empty paragraph?

**Comment by recena** [comment]
@aheritier I have to try it in my local environment. I missed some javadoc. Nice work!

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java (hunk 3 above)
Comment (ndeloof) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java]: disagree. This is a one line null check, not expected to do more. Using bracket style reduces readability.

Comment (recena) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java]: Why don't you do the same here?

```
if (jenkins == null) {
    throw new IllegalStateException("Jenkins has not been started, or was already shut down");
}
```

File: src/main/java/com/cloudbees/simplediskusage/JobDiskItem.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/simplediskusage/JobDiskItem.java]: +1. It's pretty sensitive for public interfaces

Comment (ndeloof) [inline src/main/java/com/cloudbees/simplediskusage/JobDiskItem.java]: why ? I mean, how does this help to get such unconstrained parameterized parameter type ?

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java (hunk 3 above)
Comment (aheritier) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsageInitializer.java]: yes for just a return I prefer to keep it like this

File: src/main/java/com/cloudbees/simplediskusage/JobDiskItem.java (hunk 1 above)
Comment (aheritier) [inline src/main/java/com/cloudbees/simplediskusage/JobDiskItem.java]: will be fixed.

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java (hunk 5 above)
Comment (aheritier) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java]: fixed

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java (hunk 6 above)
Comment (aheritier) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java]: fixed

File: src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java (hunk 7 above)
Comment (aheritier) [inline src/main/java/com/cloudbees/simplediskusage/QuickDiskUsagePlugin.java]: fixed

File: src/main/resources/com/cloudbees/simplediskusage/QuickDiskUsagePlugin/index.jelly (hunk 8 above)
Comment (aheritier) [inline src/main/resources/com/cloudbees/simplediskusage/QuickDiskUsagePlugin/index.jelly]: To have a white space between the table and the div.... Crappy :-)

Comment (recena) [inline src/main/resources/com/cloudbees/simplediskusage/QuickDiskUsagePlugin/index.jelly]: In the next `<div>`you have a `style` attribute... `margin-top: 10px` :dizzy_face:

Comment (aheritier) [inline src/main/resources/com/cloudbees/simplediskusage/QuickDiskUsagePlugin/index.jelly]: :-) fixed

**Pull request merged**